### PR TITLE
Compile and test

### LINKED ISSUE #11: Implement flake8 linting on CI server
Add flake8 linting on CI server in order to handle compilation.

### LINKED ISSUE #16: Make CI server run pytest
Add running of the tests folder using pytest inside the server.

### PR DESCRIPTION
**Pull request by CalleFreme**
Close #11 and close #16

### CODE REVIEW

File: src/ci_utils.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import subprocess
-from subprocess import check_output
+from subprocess import check_output  # Remove?
```
Comment (majate) [inline src/ci_utils.py:2]: Yes, this is unused and can be removed

**Review by jenniferlindberg — APPROVED** [review]
Looks good, however we should add some more comments in a later cleanup!

**Review by majate — COMMENTED** [review]
Look great! Maybe need some docstrings to the functions?

File: src/ci_utils.py [hunk 2]
```diff
@@ -34,3 +37,25 @@ def setup_repo(branch):
...
+def log_to_file(file, branch, sha, p):
+    with open("../" + file, 'a+') as log:
+        # print meta data about test run to the log
+        log.write("\n{date} :: {branch} -- {sha}:\n".format(date=datetime.now(), branch=branch, sha=sha))
+        # print the test output to the log
+        log.write(p.stdout.decode('utf-8'))
+
+
+def run_compile(branch, sha):
+    sub_proc = subprocess.run(['python{}'.format(sys.version[:3]), '-m', 'flake8', '--ignore=E501', '../git_repo/'], capture_output=True)
```
Comment (omidhzr) [inline src/ci_utils.py:51]: this is very nice! I liked it!

**Review by omidhzr — APPROVED** [review]
I have ran it on my machine and it seems to work just fine as it should, good job! :)

**Review by andnil5 — APPROVED** [review]
LGTM

**Pull request merged**